aiida/orm/computer.py

```diff
@@ -8,39 +8,708 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 from __future__ import absolute_import
-from aiida.orm.implementation import Computer
-from aiida.orm.utils import BackendDelegateWithDefault
+import abc
+import logging
+import os
+import six
 
+from aiida import transport, scheduler
+from aiida.common import exceptions
+from .backend import Collection, CollectionEntry
 
-from aiida.backends import settings
-from aiida.backends.profile import BACKEND_DJANGO, BACKEND_SQLA
+__all__ = ['Computer', 'ComputerCollection']
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ComputerCollection(Collection):
+    """The collection of Computer entries."""
+
+    @abc.abstractmethod
+    def create(self, name, hostname, description='', transport_type='', scheduler_type='', workdir=None,
+               enabled_state=True):
+        """
+        Create new a computer
+
+        :return: the newly created computer
+        :rtype: :class:`aiida.orm.Computer`
+        """
+        pass
+
+    def get(self, id=None, name=None, uuid=None):
```

Comment:
I think we should avoid to have another implementation of loading Computer entities. There already is the `ComputerEntityLoader` in `aiida.orm.utils.loaders` that also supports loading through partial uuids and without dashes. When other ORM classes will be ported to this paradigm, ideally their `get` methods all have the same signature and just call through to the entity loaders

Reply:
I'm not sure I agree.  I rather think that the hierarchy should be that the collections acts as near direct reflections of what's in the backend and don't necessarily provide any niceties regarding partial matches, etc, and therefore form the low-level interface.  Then of course it's good to have things on top of this that do more user facing operations.  They could _maybe_ even be sepearate appropriately named methods in the collection class.

Reply:
I've left it this way for now.  I reckon it'll require a bit more discussion to get an agreemnet on where to put entity loaders.

Reply:
Aye agreed, let's leave that for another time then